File: geekbang/algorithm-models/2.4-双指针夹逼模板/twoSum.go
package main

// Go
// LeetCode 167 两数之和 - 输入有序数组
func twoSum(numbers []int, target int) []int {
	j := len(numbers) - 1
	for i := range numbers {
		for i < j && numbers[i]+numbers[j] > target {
			j--
		}
		if i < j && numbers[i]+numbers[j] == target {
			return []int{i + 1, j + 1}
		}
	}
	return nil
}

File: geekbang/algorithm-models/2.4-双指针夹逼模板/twoSum.py
# Python
# LeetCode 167 两数之和 - 输入有序数组
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        j = len(numbers) - 1
        for i in range(len(numbers)):
            while i < j and numbers[i] + numbers[j] > target:
                j -= 1
            if i < j and numbers[i] + numbers[j] == target:
                return [i + 1, j + 1]
        return
